app/src/main/java/com/example/habitsmasher/HabitList.java

```diff
@@ -95,19 +95,19 @@ public void addHabitLocal(Habit habit) {
         _habits.add(habit);
     }
 
-    /**
-     * Method that edits a Habit in the specified pos in the local HabitList
-     * @param newTitle new title of habit
-     * @param newReason new reason of habit
-     * @param newDate new date of habit
-     * @param pos position of habit
-     */
-    public void editHabitLocal(String newTitle, String newReason, Date newDate, int pos) {
-        Habit habit = _habits.get(pos);
-        habit.setTitle(newTitle);
-        habit.setReason(newReason);
-        habit.setDate(newDate);
-    }
```

Comment:
This method wasn't even being used.

Reply:
editHabitLocal should be used in the editHabitInDatabase method, like addHabitLocal is used in the addHabitToDatabase. I don't know how I managed to miss that it wasn't even in the method.

Reply:
Hmmmm so probably don't delete it then?

Reply:
Well, the edit works perfectly fine without it. I don't think this method is actually needed. Every time we navigate **back** to the habit list, we **fetch and populate** the local habit list with the values in the db. I can add it back if needed, but there is no bug here... Everything works as expected, this method wasn't used outside of tests.

Reply:
It works because the change still goes through to _snapshots, so visually it appears fine, however, the edit never actually happens in the context of the local HabitList. So we need editHabitLocal to actually ensure the changes are applied to the local HabitList.

And it doesn't populate the HabitList every time we navigate to the HabitListFragment, it only does it once, when the user initially navigates to it. The **if (habitList.getHabitList().isEmpty)** conditional enforces this, and we need this because if we don't, then every time we navigate back to the HabitList, the contents of snapshots are appended to the end of HabitList.

Reply:
Alright, you know more about this workflow than I do. I will add it back, but I will leave it up to you (@jaskim9824) to add it back to where it needs to be in the first place. Since you're the most familiar with this part of the code.